### DISCUSSION TITLE
[feature] make queryClient is an option of  `useQuery`

### DISCUSSION THREAD

**Discussion by NewFuture** [TanStack/query#3234] [General] [answered]
Pass queryClient in `useQuery`

```ts
useQuery({});// use Query Client  in context form provider
useQuery({queryClient });  // ignore the context and use the queryClient  in arguments 
```

**Comment by TkDodo** [comment]
what would you need it for please?
see also: https://github.com/tannerlinsley/react-query/issues/3151

**Comment by TkDodo** [comment, marked as answer]
oh, I misunderstood, sorry. seems like you want:
- https://github.com/tannerlinsley/react-query/pull/2548

**Reply by NewFuture** [reply]
yeah, it seems this feature! 😄

**Reply by TkDodo** [reply]
it will likely land in v4
